docs/spl-docs/bin-command.md

```diff
@@ -0,0 +1,219 @@
+
+#### \<timescale\>
+**Syntax:** `<subsecond> | <second> | <minute> | <hour> | <day> | <week> | <month> | <quarter> | <year>`\
+**Description:** \
+`<timescale>` is a `<string>` that specifies the unit of time.\
+The magnitude of time represented by `<int><subsecond>` must be less than 1 second and evenly divisible by 1 second.\
+For `<month>`, the only permissible `<int>` values are `1, 2, 3, 4, 6, 12`.\
+For `<quarter>`, the only permissible `<int>` values are `1, 2, 4`.
+
+| **Unit** | **Strings used to specify Unit** |
+| --------- | ----------------------------- |
+| `<subsecond>` | millisecond (ms), centisecond (cs), decisecond (ds) |
+| `<second>` | seconds, second, secs, sec, s  |
+| `<minute>` | minutes, minute, mins, min, m |
+| `<hour>` | hours, hour, hrs, hr, h |
+| `<day>` | days, day, d |
+| `<week>` | weeks, week, w |
+| `<month>` | months, month, mon |
+| `<quarter>` | quarters, quarter, qtrs, qtr, q |
+| `<year>` | years, year, yrs, yr, y |
+
+
+## How does bin work?
+The `<span>` option always takes priority. All other options are ignored when `<span>` is explicitly specified.
+
+If not specified, the span is derived based on the data distribution of the `<field>`.\
+If `<minspan>` is specified, the derived span must follow this property.\
+In the absence of an explicitly specified `<span>`, the derived span must strictly adhere to the maximum number of bins specified by `<bins>` or its default value.
+
+**For Numerical Data:**\
+`<span-length>` or `<log-span>` can be used to specify the exact span of the bin using `<span>` option.\
+The derived `<span-length>` for bins will always be in powers of 10.
+
+**For Time-Based Data:**\
+Currently, SigLens only supports time discretization on the default `timestamp` field.\
+`<span-length>` can be used to specify the exact span of the bin using `<span>` option. The minimum span length that meets all the specified properties will be decided.\
+All derived `<span-length>` will be aligned to the UTC timezone by default.\
+The derived `<span-length>` are fixed and are listed below in ascending order:
+- 1 second
+- 10 seconds
+- 30 seconds
+- 1 minute
+- 5 minutes
+- 10 minutes
+- 30 minutes
+- 1 hour
+- 1 day
+- 1 month
+
+The `<bins>` option will be ignored in this case if it is not possible to accommodate all the data into the maximum number of bins specified by the largest derived span, i.e., **1 month**.
+
+
+## Examples
+
+The following example creates bins based on the `timestamp` field and aligns the bins to the UTC time of 1720713887000.
+```
+...| bin aligntime=1720713887000 timestamp
+```
+
+
+## Use-Case Examples
+
+**Analyzing Event Latency in Real-Time**
+
+**Problem:** The challenge is to understand how the latency of events fluctuates over very short intervals, specifically on a second-by-second basis. This analysis is crucial for identifying performance bottlenecks in real-time systems where even minor delays can impact user experience or system efficiency.
```

Comment:
Since you have multiple examples, it would be good to either label them (e.g., `**Example 1: Analyzing Event Latency in Real-Time**), or potentially make them a header like
```
### Analyzing Event Latency in Real-Time
```

Just to help separate the start of a new example from the end of the previous example

Reply:
nit: the examples you have seem pretty similar to each other. It would be good to have a little more variety (e.g., show an example where using a log span size is helpful)

Reply:
addressed in the latest commit